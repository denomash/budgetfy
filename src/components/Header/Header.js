import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';

import { startLoggout } from '../../actions/auth';

export const Header = ({ startLoggout }) => {
  return (
    <div>
      <h1>Budgetify</h1>
      <ul>
        <li>
          <NavLink exact to="/dashboard" activeClassName="is-active">
            Go Home
          </NavLink>
        </li>
        <li>
          <NavLink to="/add" activeClassName="is-active">
            Add expense
          </NavLink>
        </li>
        <li>
          <NavLink to="/help" activeClassName="is-active">
            Help
          </NavLink>
        </li>
      </ul>
      <button onClick={startLoggout}>Logout</button>
      <br />
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  startLoggout: () => dispatch(startLoggout())
});

export default connect(
  null,
  mapDispatchToProps
)(Header);
