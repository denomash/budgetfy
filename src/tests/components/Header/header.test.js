import React from 'react';
import { shallow } from 'enzyme';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { Header } from '../../../components/Header/Header';

test('Should render header correctly', () => {
  const wrapper = shallow(<Header startLoggout={() => {}} />);

  expect(wrapper).toMatchSnapshot();

  // expect(wrapper.find('h1').length).toBe(1);
  // const renderer = new ReactShallowRenderer();

  // renderer.render(<Header />);

  // expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('Should call startLogout on button click', () => {
  const startLoggout = jest.fn();
  const wrapper = shallow(<Header startLoggout={startLoggout} />);

  wrapper.find('button').simulate('click');

  expect(startLoggout).toHaveBeenCalled();
});
