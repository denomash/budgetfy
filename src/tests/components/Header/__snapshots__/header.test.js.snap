// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<div>
  <h1>
    Budgetify
  </h1>
  <ul>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        exact={true}
        to="/"
      >
        Go Home
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/add"
      >
        Add expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/edit"
      >
        Edit Expense
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="is-active"
        ariaCurrent="true"
        to="/help"
      >
        Help
      </NavLink>
    </li>
  </ul>
</div>
`;
